### PR TITLE
Group sensor type data

### PR DESCRIPTION
**Pull request by nerdlet**
## Description
Fixes # (issue)
- [x] Group data by sensor_types

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

screencapture-localhost-3000-dashboard-index-html-2021-06-24-07_35_12

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: public/dashboard/index.html [hunk 1]
```diff
@@ -110,267 +110,200 @@
...
+                        ticks: {
+                          beginAtZero: true,
+                        },
+                    }]
+                  },
+                  tooltips: {
+                  }
+                }
+              });
+              </script>
```
Comment (kilemensi) [inline public/dashboard/index.html:235]: What's happening here?

**Review by kilemensi — COMMENTED** [review]
@nerdlet , our data is for cities but we need to do our computations per county so the first task is to make sure we we've grouped city to nodes and have data for county. I don't see this step.

File: public/js/charts/sensorCoverage.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    for (let i = 0; i < arr.length; i += 1) {
+      if (arr[i] instanceof Array) {
+        return getAllSensorTypes(arr[i]);
+      }
+      return Object.fromEntries(
+        Object.entries(arr[i]).filter(([key]) => key === 'sensordatavalues')
+      );
+    }
+    return null;
+  }
```
Comment (kilemensi) [inline public/js/charts/sensorCoverage.js]: Is this being used anywhere?

File: public/js/sensorsChart.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+const sensorsChart = document.getElementById("sensorsChart").getContext("2d");
```
Comment (kilemensi) [inline public/js/sensorsChart.js]: Which chart is this?

File: public/js/charts/sensorCoverage.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            h[obj.value_type] = (h[obj.value_type] || []).concat(obj);
+            return h; 
+          }, {});
+          const sensorTypes = Object.keys(aLLNodesReduced).map(key => {
+            return {
+                sensor: key, 
+                sensorAvg : Math.round(aLLNodesReduced[key].reduce((a, b) => a + (Number(b.value) || 0), 0)/aLLNodesReduced[key].length),
+            } 
+         });
+         const sensorCoverageData = sensorTypes?.filter(item  => item.sensor !== "timestamp" && item.sensor !== "height" && item.sensor !== "lon" && item.sensor !== "lat").map(item => item.sensorAvg);
```
Comment (kilemensi) [inline public/js/charts/sensorCoverage.js]: You can chain these calls

**Review by kilemensi — COMMENTED** [review]
👍🏾 

----

Lint/format the `sensorCoverage` file

File: public/js/charts/sensorCoverage.js (hunk 2 above)
Comment (nerdlet) [reply public/js/charts/sensorCoverage.js]: @kilemensi Nope forgot that there. Will remove it

File: public/js/sensorsChart.js (hunk 3 above)
Comment (nerdlet) [reply public/js/sensorsChart.js]: @kilemensi Um, Is the sensors coverage chart. I'll probably rename it  to match its  equivalent js files

Comment (kilemensi) [reply public/js/sensorsChart.js]: There is already sensors coverage chart in `public/js/charts/sensorsCoverage.js` @nerdlet

**Review by gertie-sheshe — COMMENTED** [review]
Your indentation settings seem to be different than what we use @nerdlet

**Comment by nerdlet** [comment]
> Your indentation settings seem to be different than what we use @nerdlet
@gertie-sheshe Ahhh, okay....must be the formatting I applied. I'll fix it

File: public/js/sensorsChart.js (hunk 3 above)
Comment (nerdlet) [reply public/js/sensorsChart.js]: @kilemensi Ahhh, okay...deleted it

File: public/js/charts/sensorCoverage.js (hunk 4 above)
Comment (nerdlet) [reply public/js/charts/sensorCoverage.js]: @kilemensi Just pushed a fix on this. Could let me know if this works

**Review by kilemensi — APPROVED** [review]
🚀 

----

We can still improve the code but I guess it's good enough for now. Formatting though, don't you have ESLint plugin installed in your VS Code? It should automatically pick up project config from there.

**Comment by nerdlet** [comment]
> 🚀
> 
> We can still improve the code but I guess it's good enough for now. Formatting though, don't you have ESLint plugin installed in your VS Code? It should automatically pick up project config from there.

@kilemensi Yeah, didn't have eslint installed. Should be good now

**Pull request merged**